habitat/datasets/rearrange/sim_utilities.py

```diff
@@ -0,0 +1,838 @@
+            if sub_config_key.startswith("receptacle_"):
+                sub_config = stage_user_attr.get_subconfig(sub_config_key)
+                # this is a receptacle, parse it
+                assert sub_config.has_value("position")
+                assert sub_config.has_value("scale")
+                up = (
+                    None
+                    if not sub_config.has_value("up")
+                    else sub_config.get("up")
+                )
+
+                receptacle_name = (
+                    sub_config.get("name")
+                    if sub_config.has_value("name")
+                    else sub_config_key
+                )
+                rotation = sub_config.get("rotation")
+
+                receptacles.append(
+                    Receptacle(
+                        name=receptacle_name,
+                        bounds=mn.Range3D.from_center(
+                            sub_config.get("position"),
+                            sub_config.get("scale"),
+                        ),
+                        rotation=rotation,
+                        up=up,
+                    )
+                )
+
+    # rigid objects
+    for obj_handle in obj_mgr.get_object_handles():
+        obj = obj_mgr.get_object_by_handle(obj_handle)
+        user_attr = obj.user_attributes
+
+        for sub_config_key in user_attr.get_subconfig_keys():
+            if sub_config_key.startswith("receptacle_"):
+                sub_config = user_attr.get_subconfig(sub_config_key)
+                # this is a receptacle, parse it
+                assert sub_config.has_value("position")
+                assert sub_config.has_value("scale")
+                up = (
+                    None
+                    if not sub_config.has_value("up")
+                    else sub_config.get("up")
+                )
+
+                receptacle_name = (
+                    sub_config.get("name")
+                    if sub_config.has_value("name")
+                    else sub_config_key
+                )
+                receptacles.append(
+                    Receptacle(
+                        name=receptacle_name,
+                        bounds=mn.Range3D.from_center(
+                            sub_config.get("position"),
+                            sub_config.get("scale"),
+                        ),
+                        up=up,
+                        parent_object_handle=obj_handle,
+                    )
+                )
+
+    # articulated objects #TODO: merge with above
+    for obj_handle in ao_mgr.get_object_handles():
+        obj = ao_mgr.get_object_by_handle(obj_handle)
+        user_attr = obj.user_attributes
+
+        for sub_config_key in user_attr.get_subconfig_keys():
+            if sub_config_key.startswith("receptacle_"):
+                sub_config = user_attr.get_subconfig(sub_config_key)
+                # this is a receptacle, parse it
+                assert sub_config.has_value("position")
```

Comment:
This logic to parse Receptacle fields from a subconfig is mostly duplicated three times in this function. Can we consolidate?